valhalla/mjolnir/graphbuilder.h

```diff
@@ -55,6 +69,71 @@ class GraphBuilder {
                                  bool tc);
 };
 
+// The tile manifest is a JSON-serializable index of tiles to be processed during the build stage of
+// valhalla_build_tiles'. It can be used to distribute shard keys when building tiles with
+// parallelized, distributed batch processing. For example, a workflow orchestrator can partition
+// and distribute this manifest to workers in a 'build' stage worker pool.
+//
+// This file written out during 'constructedges' and a prerequisite for the 'build' stage run by
+// valhalla_build_tiles.
+//
+// Example manifest :
+//
+// {
+//   "tiles": [
+//     {
+//       "node_index": 0,
+//       "graphid": {
+//         "value": 5970538,
+//         "id": 0,
+//         "tile_id": 746317,
+//         "level": 2
+//       }
+//     }
+//   ]
+// }
+struct TileManifest {
```

Comment:
this struct and its implementation should live inside of `mjolnir/util` since that is where it is used. i would even prefer it were completely a hidden implemntation detail inside of the util.cc. i know that means that unit testing it has to be via util::BuildTiles/util::ConstructEdges but we have to test those anyway? i guess your test will make a manifest and load it and then split it in two and buildtiles twice to confirm you get the same as when you dont do that?

Reply:
I was thinking about two tests: one around serialization, and another around what you're suggesting. Also, a test to verify behavior if we actually build a subset of tiles.

I can move it to util!